### ISSUE TITLE
Tileable texture from Stable Diffusion

### ISSUE THREAD

**Issue by hlky** [hlky/stable-diffusion-webui#30]
Someone requested [Tileable textures](https://gitlab.com/-/snippets/2395088) be implemented, this code is for diffusers codebase, needs implementing for stable-diffusion codebase

**Comment by KawaNae** [comment]
I simply inserted this partial code from [here](https://gitlab.com/-/snippets/2395088) in line 34 of optimized_txt2img.py and got Tileable texture scripts for stable-diffusion codebase. (I'm new to python so I could be wrong on this.)

```python
def patch_conv(klass):
	init = klass.__init__
	def __init__(self, *args, **kwargs):
		return init(self, *args, **kwargs, padding_mode='circular')
	klass.__init__ = __init__

for klass in [torch.nn.Conv2d, torch.nn.ConvTranspose2d]:
	patch_conv(klass)
```

![image](https://user-images.githubusercontent.com/78061734/187039812-545c71a0-90be-4c98-98f6-1f9a1fa52744.png)

**Comment by FuouM** [comment]
https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html

from the documentation, the "padding mode" have several options:
```
padding_mode (string, optional) – 'zeros', 'reflect', 'replicate' or 'circular'. Default: 'zeros'
```

the default mode is `zeros` so I think it would be wise to have a toggle to switch between them
